### PR TITLE
Fix deferred tests

### PR DESCRIPTION
**Pull request by porter-stripe**
## Summary
- A few deferred tests were failing and were disabled in the deploy yesterday
- One was failing due to billing details not being present for Apple Pay. I'm guessing this passed originally because a test before it may have ran and added the billing details but the test order is random causing it to fail intermediately.
- The payment methods returned from elements session randomly changed to just `card`.  It's pretty unclear why, but I changed it to just card and cash app which fixes the issue but we may want to just check if the returned payment methods are not empty since the behavior seems flaky. Something we should follow up on.

## Motivation
Deferred workflow

## Testing
N/A

## Changelog
N/A

### CODE REVIEW

File: Stripe/StripeiOSTests/PaymentSheet+APITest.swift [hunk 1]
```diff
@@ -218,8 +218,8 @@ class PaymentSheetAPITest: XCTestCase {
         let confirmExpectation = XCTestExpectation(description: "Confirm deferred intent")
         let callbackExpectation = XCTestExpectation(description: "Confirm callback invoked")
 
-        let types = ["ideal", "card", "bancontact", "sofort"]
-        let expected: [STPPaymentMethodType] = [.card, .iDEAL, .bancontact, .sofort]
+        let types = ["card", "cashapp"]
+        let expected: [STPPaymentMethodType] = [.card, .cashApp]
```
Comment (porter-stripe) [inline Stripe/StripeiOSTests/PaymentSheet+APITest.swift:222]: Maybe we just check if the returned payment methods are not empty?

File: Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift [hunk 2]
```diff
@@ -799,6 +798,38 @@ extension PaymentSheetUITest {
         XCTAssertTrue(cardButton.waitForExistence(timeout: 10.0))
         cardButton.forceTapElement()
 
+        // Fill out billing details if required
+        let addBillingDetailsButton = applePay.buttons["Add Billing Address"]
```
Comment (yuki-stripe) [inline Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift:802]: do we do this elsewhere?  I'm not finding it.

File: Stripe/StripeiOSTests/PaymentSheet+APITest.swift (hunk 1 above)
Comment (yuki-stripe) [reply Stripe/StripeiOSTests/PaymentSheet+APITest.swift:222]: that also makes sense

**Review by yuki-stripe — APPROVED** [review]

File: Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift (hunk 2 above)
Comment (porter-stripe) [reply Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift:802]: I can't find it anywhere, but that was my best guess as to how it passed in the first place...

Comment (porter-stripe) [reply Example/PaymentSheet Example/PaymentSheetUITest/PaymentSheetUITest.swift:802]: Somehow the simulator had a billing address attached to Apple Pay when CI passed

File: Stripe/StripeiOSTests/PaymentSheet+APITest.swift (hunk 1 above)
Comment (porter-stripe) [reply Stripe/StripeiOSTests/PaymentSheet+APITest.swift:222]: Going to leave as-is for now since we know the root cause

**Pull request merged**